homeassistant/components/fjaraskupan/binary_sensor.py

```diff
@@ -0,0 +1,87 @@
+"""Support for sensors."""
+from __future__ import annotations
+
+from fjaraskupan import Device, State
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from . import DeviceState, async_setup_entry_platform
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors dynamically through discovery."""
+
+    def _constructor(devicestate: DeviceState):
+        return [
+            GreaseFilterSensor(
```

Comment:
I'd recommmend to instead create one binary sensor entity class and define two entity descriptions for the two different cases instead. We can add a lambda for the correct filter attribute access. Eg:

https://github.com/home-assistant/core/blob/fe6c8967540d4a26f769b35608fc232841e14399/homeassistant/components/airly/const.py#L81-L88